scipy/signal/spectral.py

```diff
@@ -766,11 +770,15 @@ def _spectral_helper(x, y, fs=1.0, window='hann', nperseg=256,
                 pad_shape[-1] = x.shape[-1] - y.shape[-1]
                 y = np.concatenate((y, np.zeros(pad_shape)), -1)
 
+    # parse window; if array like, then set nperseg = win.shape
+    win, nperseg = _triage_segments(window, nperseg)              
+        
     # X and Y are same length now, can test nperseg with either
     if x.shape[-1] < nperseg:
         warnings.warn('nperseg = {0:d}, is greater than input length = {1:d}, '
                       'using nperseg = {1:d}'.format(nperseg, x.shape[-1]))
         nperseg = x.shape[-1]
+        win, nperseg = _triage_segments(window, nperseg) 
```

Comment:
It looks like this would not catch the case where the window's length is greater than the signal length. 

If `nperseg=None` and the window has greater length than the signal, we should error out, since there's no good way to adapt the window to the signal. There should be test for this case as well.